### ISSUE TITLE
[Operator V2] Access to DS Projects fails if DS Pipelines are disabled from DSC

### ISSUE THREAD

**Issue by bdattoma** [opendatahub-io/opendatahub-operator#519]
**Describe the bug**
When trying to access a Data Science Project using a non-cluster-admin user, we get a permission error related to pipelines-definition, which prevents us to use DS Projects UI

**To Reproduce**
1. Install Operator V2
2. Enable Dashboard and disable DS Pipelines in the DataScienceCluster object
3. Log In Dashboard UI with a user who has no cluster admin permissions
4. create a DS Project rom Dashboard UI
5. Access the DS Project from Dashboard UI
6. See error

**Expected behavior**
No errors, user can use the DS Project UI page

**Screenshots**
![image](https://github.com/opendatahub-io/opendatahub-operator/assets/88311595/1a519e03-ba89-4ff2-b42d-767e4a1b8d2d)

**Additional context**

- OpenShift Pipelines 1.11.1 operator is installed 
- DSC has dashboard,kserve and workbenches enabled only
- enabling datasciencepipelines solves the issue

**Comment by zdtsw** [comment]
is this the expected behavior in v1 operator? 
from the error message and the code, we do not grant permission in v2 explicitly in this case, but we need confirm if this should be the case, because in v1 we might open the permission a bit too wild.

**Comment by bdattoma** [comment]
>is this the expected behavior in v1 operator?


This is not a case for operator v1 because we cannot disable components by using DSC


DS projects have more features than Pipelines, so I believe a user of V2 operator can choose to enable workbenches and not pipelines. if that the case, the DS project must be accessible

**Comment by zdtsw** [comment]
so the real solution is we need to get the rules on datasciencepipelinesapplications with 
```
    rbac.authorization.k8s.io/aggregate-to-admin: "true"
    rbac.authorization.k8s.io/aggregate-to-edit: "true" 
```
so the non-admin users can  get admin privileges that will have the permissions specified in this ClusterRole.

purely with PR is not enough #529

**Comment by zdtsw** [comment]
and seems like we have several components actually are setting aggregate-to-admin for different clusterroles.

**Comment by zdtsw** [comment]
@harshad16 @atheo89 workbench and @lucferbux @andrewballantyne UI teams, could you help me to identify the issue here: if it is something from the workbench or dashboard or even operator?

TL;DR:
in v2 build, `non-admin user` get error message when access any `Data Science Projects` if the `DSPA` component is `Removed`. 

offline discussed with Berto, this is a wrong behavior, that ODH operator does not need to have `DSPA` component `Managed` in order for non-admin user to view Data Science Projects.
 
in v1, since all components are by default enabled => ClusterRole with aggragated admin permission are added, but in v2, we only create such if the component is Managed.

So the question is,  why it requires such permission if non-admin user wants to only access Data Science Projects.

**Comment by bdattoma** [comment]
the workaround seems to be disabling pipelines from OdhDashboardConfig instance: `disablePipelines: false`

**Comment by harshad16** [comment]
This seems like UI people would be able to answer better than the notebooks-teams.
some follow up question , that might help with the discussion:
- Does removing the workbench component , cause the same issue ?
- Does removing the model serving component, cause the same issue?

probably as current in the data science project instance: workbenches, data science pipeline and model serving
are provided, removing them breaks the flow of the data science project.

**Comment by bdattoma** [comment]
@harshad16 I tried removing model serving component, and the behavior is different: user is able to access their DS Project, but an error msg is rendered in the model serving section (see the screenshot below).

![image](https://github.com/opendatahub-io/opendatahub-operator/assets/88311595/12497cba-fbbe-4d6d-9a59-6bb8b8221b76)

**Comment by harshad16** [comment]
Seems like then the close relation is only with data science pipeline, perhaps as you mentioned the workaround
that is the content causing this close bond. 
UI team would be the best to answer this.

**Comment by andrewballantyne** [comment]
This definitely seems like a Dashboard issue. Interesting state of K8s to give a 403 error if the CRD does not exist for basic users.

/transfer odh-dashboard

Hopefully the bot exists in this repo 🙂

**Comment by bdattoma** [comment]
thank you all for looking into it :)
